### PR TITLE
WIP: Finding enriched signatures for a given set of genes

### PR DESCRIPTION
**Pull request by rzelayafavila**
Changes for https://github.com/greenelab/adage-server/issues/153

### CODE REVIEW

File: interface/src/app/signature/enriched_signatures.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+      var m = userSelectedGenes.length;
+
+      var pValueArray = [];
+
+      var signatureIds = Object.keys(signatures);
+
+      for (var i = 0; i < signatureIds.length; i++) {
+        var signatureId = signatureIds[i];
+        var genes = signatures[signatureId];
+        var n = genes.length;
```
Comment (dongbohu) [inline interface/src/app/signature/enriched_signatures.js]: @rzelayafavila: Is `n` the total number of heavy genes associated with `signatureId`?

File: interface/src/app/signature/enriched_signatures.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      significantSignatureArray.sort(function(a, b) {
+        return a.pValue - b.pValue;
+      });
+
+      return significantSignatureArray;
+    };
+
+
+    Participation.get(
+      {'mlmodel': self.modelInUrl, 'gene__in': $stateParams.genes, 'limit': 0},
```
Comment (dongbohu) [inline interface/src/app/signature/enriched_signatures.js]: @rzelayafavila: If `n` on line 104 (see my comment above) is the total number of heavy genes in a certain signature (not limited to the genes in the URL), why we need `gene__in` parameter here? I feel that this parameter should be removed.

File: interface/src/app/signature/enriched_signatures.js (hunk 1 above)
Comment (rzelayafavila) [reply interface/src/app/signature/enriched_signatures.js]: Hi @dongbohu - Yes, it is meant to be the total number of genes that have participation in a signature (for a specific type of participation, can be high weight genes or another type of participation).

File: interface/src/app/signature/enriched_signatures.js (hunk 2 above)
Comment (rzelayafavila) [reply interface/src/app/signature/enriched_signatures.js]: Ah I see - I think that is true if you want to get ALL participations back from the database and then figure out which participations (and therefore signatures) are relevant to this analysis via JavaScript here.

**Pull request closed without merging**